lib/floe/workflow/runner.rb

```diff
@@ -12,35 +12,21 @@ def initialize(_options = {})
 
       @runners = {}
       class << self
-        # deprecated -- use Floe.set_runner instead
-        def docker_runner=(value)
-          set_runner("docker", value)
+        def register_scheme(scheme, klass_or_proc)
+          @runners[scheme] = klass_or_proc
         end
 
-        # see Floe.set_runner
-        def set_runner(scheme, name_or_instance, options = {})
-          @runners[scheme] =
-            case name_or_instance
-            when "docker", nil
-              Floe::Workflow::Runner::Docker.new(options)
-            when "podman"
-              Floe::Workflow::Runner::Podman.new(options)
-            when "kubernetes"
-              Floe::Workflow::Runner::Kubernetes.new(options)
-            when Floe::Workflow::Runner
-              name_or_instance
-            else
-              raise ArgumentError, "docker runner must be one of: docker, podman, kubernetes"
-            end
+        private def resolve_scheme(scheme)
+          runner = @runners[scheme]
+          runner = @runners[scheme] = @runners[scheme].call if runner.is_a?(Proc)
+          runner
```

Comment:
The old `set_runner` wasn't robust enough. You will note that `providers-workflows` is basically this but with a bunch of special cases for the options hash.

Is it possible to move some of the provider special logic into here? (possibly getting rid of that?)

Reply:
> Is it possible to move some (all?) of the provider special logic into here?

Some, yes, however the _options_ are going to be significantly different so you'll still need that case statement in the workflows plugin.